inbound-saml tries to create entity with no token on http

**Describe the bug**

While using `http` and creating a new trusted IDP through metadata URL, `inbound-saml` attempts to retrieve an authorization token from the token endpoint, which returns an error. `Inbound-saml` does not handle this and proceeds to try to create an entity with no token, leading to a `Failed to find entry` exception, which causes a status code `401` in `inbound-saml`.

**To Reproduce**

1. Install Gluu Server >= 4.4.0
2. Side install `inbound-saml` as described [here](https://github.com/GluuFederation/inbound-saml/blob/master/docs/getting_started.md).
3. When setting up Persistence API environment variables, ensure `INBOUND_SAML_OXTRUST_API_TOKEN_URL` is using `http`.
4. Start the SP Proxy Server in dev mode
5. Use `https://<hostname>/inbound-saml/trust-relation/metadata` to send a POST request to create a new trusted IDP ([example](https://github.com/GluuFederation/inbound-saml/blob/docs/openapi.yml))

**Expected behavior**
Returns status code `201`, new trusted IDP is added.

**Actual behavior**
Returns status code `500`: internal server error.

**Logs/Output**
A look at the SP Proxy Server console output reveals ` [ERROR]    Request failed with status code 401`. In addition, `oxtrust.log` reveals the following:

```
TRACE [qtp952562199-1073] [org.gluu.oxtrust.auth.uma.UmaPermissionService] (UmaPermissionService.java:189) - Validating RPT, resourceId: <client ID>, scopeIds: [oxtrust-api-write], authorization: Bearer undefined

```

**Additional context**
`inbound-saml` may need to either validate the bearer token (not accepting `undefined`) or raise an error for `http` connections.

--- a/docs/getting_started.md
+++ b/docs/getting_started.md
@@ -51,7 +51,7 @@ Data layers calls oxTrust REST API that handles persistence to multiple backends
 | `INBOUND_SAML_OXTRUST_API_HOST`          | The oxTrust Api Host (i.e. `mysubdomain.mydomain.org`)                                | None                                  |
 | `INBOUND_SAML_OXTRUST_CLIENT_ID`         | OxTrust **API RP** client ID. Open ID Client ID for API Ressource Provider.           | None                                  |
 | `INBOUND_SAML_OXTRUST_API_COMPLETE_PATH` | OxTrust API Complete Path. (without the first forward slash)                          | `identity/restv1/api/v1/inbound-saml` |
-| `INBOUND_SAML_OXTRUST_API_TOKEN_URL`     | OxTrust API Token Url. URL to get token fro oxAuth for accessing the persistence api. | None                                  |
+| `INBOUND_SAML_OXTRUST_API_TOKEN_URL`     | OxTrust API Token Url. URL to get token fro oxAuth for accessing the persistence api. Should use secure protocol (`https`) according to Oauth2 specs | None                                  |
 | `INBOUND_SAML_OXTRUST_API_KID`           | Oxtrust **API RP** rs256 `kid` (extracted from jwks)                                  | None                                  |
 | `INBOUND_SAML_OXTRUST_API_PVK_PATH`      | Oxtrust **rs256 Private Key** extracted from api-rp jwks with above `kid`             | None                                  |
 

